core/modules/replica/DatabaseMySQL.cc

```diff
@@ -81,229 +49,67 @@ namespace replica {
-}
-
-//////////////////////////////////////////////////
-//                DoNotProcess                  //
-//////////////////////////////////////////////////
-
-DoNotProcess::DoNotProcess(std::string const& name_)
-    :   name(name_) {
-}
-
-/////////////////////////////////////////////
-//                Keyword                  //
-/////////////////////////////////////////////
-
-Keyword const Keyword::SQL_NULL {"NULL"};
-
-Keyword::Keyword(std::string const& name_)
-    :   DoNotProcess(name_) {
+unsigned long Connection::max_allowed_packet() {
+    return 4*1024*1024;
 }
 
-/////////////////////////////////////////////
-//                Function                 //
-/////////////////////////////////////////////
-
-Function const Function::LAST_INSERT_ID {"LAST_INSERT_ID()"};
-
-Function::Function(std::string const& name_)
-    :   DoNotProcess(name_) {
+Connection::Ptr Connection::open(ConnectionParams const& connectionParams) {
+    return open2(connectionParams,
+                 Configuration::databaseAllowReconnect(),
+                 Configuration::databaseConnectTimeoutSec());
 }
 
-///////////////////////////////////////////////
-//                Connection                 //
-///////////////////////////////////////////////
-
-Connection::Ptr Connection::open(ConnectionParams const& connectionParams,
-                                     bool autoReconnect,
-                                     bool autoCommit) {
-
-    Connection::Ptr ptr(new Connection(connectionParams,
-                                           autoReconnect,
-                                           autoCommit));
+Connection::Ptr Connection::open2(ConnectionParams const& connectionParams,
+                                  bool allowReconnects,
+                                  unsigned int connectTimeoutSec) {
+
+    unsigned int const effectiveConnectTimeoutSec =
+        0 == connectTimeoutSec ? Configuration::databaseConnectTimeoutSec()
+                               : connectTimeoutSec;
+    Connection::Ptr ptr(
+        new Connection(
+            connectionParams,
+            allowReconnects ? effectiveConnectTimeoutSec
+                            : 0
+        )
+    );
     ptr->connect();
     return ptr;
 }
 
 Connection::Connection(ConnectionParams const& connectionParams,
-                       bool autoReconnect,
-                       bool autoCommit)
-    :   _connectionParams(connectionParams),
-        _autoReconnect(autoReconnect),
-        _autoCommit(autoCommit),
+                       unsigned int connectTimeoutSec)
+    :   _id(++_nextId),
+        _connectionParams(connectionParams),
+        _connectTimeoutSec(connectTimeoutSec),
         _inTransaction(false),
         _mysql(nullptr),
+        _mysqlThreadId(0),
+        _connectionAttempt(0),
         _res(nullptr),
         _fields(nullptr),
         _numFields(0) {
-}
 
+    std::string const context = "Connection[" + std::to_string(_id) + "]  constructed";
```

Comment:
this string should be put inside the LOGS args, so it won't get built (consume cycles) if the log level excludes DEBUG.

Reply:
(I'll only leave this comment this once, but there are other instances of this)

Reply:
Ok, I see that this is a pattern and sometimes `context` is used in more than one LOG statement in the function. I wish there was a good way to conditionally build the string, but I can't think of anything easy off the top of my head.

Reply:
I've removed this string in a few trivial cases. However, in the majority of the remaining cases the context string is used in exceptions. Hence, I had to leave the code as is.
Besides, in those (other cases) the string is loaded with various information. Constructing the very same context multiple time in a body of each methods where it's use would result in a noticeable code duplication and increased "nose" in the code.